Policies/Kubernetes/forbidden-sysctl-profiles/rego/src.rego

```diff
@@ -2,19 +2,23 @@ package k8sazureforbiddensysctls
 
 violation[{"msg": msg, "details": {}}] {
     sysctl := input.review.object.spec.securityContext.sysctls[_].name
-    forbiddin_sysctl(sysctl)
+    forbidden_sysctl(sysctl)
     msg := sprintf("The sysctl %v is not allowed, pod: %v. Forbidden sysctls: %v", [sysctl, input.review.object.metadata.name, input.parameters.forbiddenSysctls])
 }
 
 # * may be used to forbid all sysctls
-forbiddin_sysctl(sysctl) {
+forbidden_sysctl(sysctl) {
     input.parameters.forbiddenSysctls[_] == "*"
 }
 
-forbiddin_sysctl(sysctl) {
+forbidden_sysctl(sysctl) {
     input.parameters.forbiddenSysctls[_] == sysctl
 }
 
-forbiddin_sysctl(sysctl) {
-    startswith(sysctl, trim(input.parameters.forbiddenSysctls[_], "*"))
+forbidden_sysctl(sysctl) {
+	# Escape any existing periods so they are matched exactly
+    initial := replace(input.parameters.forbiddenSysctls[_], ".", `\.`)
+    # Replace any wildcard * characters with the .+ regex wildcard
+    pattern := replace(initial, "*", `.+`)
```

Comment:
Instead of rego replacing the chars, I think its safer and predictable for user if we ask for regex itself as an input. You can take an array of sysctl regexes.

Reply:
Are you sure we want that? How will the portal deal with escaped strings? Will users be comfortable with regex? This will be a departure from current psp functionality

Reply:
i think we should keep it as what users would expect https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#setting-sysctls-for-a-pod